src/seraphis_main/scan_balance_recovery_utils.h

```diff
@@ -101,7 +101,12 @@ bool try_find_legacy_enotes_in_tx(const rct::key &legacy_base_spend_pubkey,
 * param: origin_status -
 * outparam: basic_records_in_tx_out -
 */
-bool try_find_sp_enotes_in_tx(const crypto::x25519_secret_key &xk_find_received,
+bool try_find_sp_enotes_in_tx(const crypto::x25519_secret_key &d_filter_assist,
+    const rct::key &input_context,
+    const SpTxSupplementV1 &tx_supplement,
+    const std::vector<SpEnoteVariant> &enotes_in_tx,
+    std::vector<bool>& is_auxiliary_out);
+bool try_find_sp_enotes_in_tx(const crypto::x25519_secret_key &d_filter_assist,
```

Comment:
The idea behind there being two `try_find_sp_enotes_in_tx` is that the first one does the meat of the work, but does the least amount of allocations and copying possible. All it really needs to allocate is the output vector of bools. This ideally would be used in actual production code which scans on behalf of users, but doesn't need to send out chunk data yet, and just caches some bools persistently (maybe as unsigned ints for compactness).